inc/Base/Activate.php

```diff
@@ -39,28 +47,46 @@ public static function activate() {
 	 */
 	private static function create_database() {
 		global $wpdb;
-
-		$smaily = "
-                CREATE TABLE IF NOT EXISTS `{$wpdb->prefix}smaily` (
-                `id` int(11) NOT NULL AUTO_INCREMENT ,
-                `enable` tinyint(1) DEFAULT NULL,
-                `subdomain` varchar(255) DEFAULT NULL,
-                `username` varchar(255) DEFAULT NULL,
-                `password` varchar(255) DEFAULT NULL,
-                `autoresponder` varchar(255) DEFAULT NULL,
-                `autoresponder_id` int(10) DEFAULT NULL,
-                `syncronize_additional` varchar(255) DEFAULT NULL,
-                `syncronize` varchar(255) DEFAULT NULL,
-                PRIMARY KEY (`id`)
-                ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
-                ";
-
+		// Create smaily settigs table.
+		$table_name = $wpdb->prefix . 'smaily';
+		$charset_collate = $wpdb->get_charset_collate();
+		$smaily     = "CREATE TABLE $table_name (
+				id int(11) NOT NULL AUTO_INCREMENT,
+				enable tinyint(1) DEFAULT NULL,
+				subdomain varchar(255) DEFAULT NULL,
+				username varchar(255) DEFAULT NULL,
+				password varchar(255) DEFAULT NULL,
+				autoresponder varchar(255) DEFAULT NULL,
+				autoresponder_id int(10) DEFAULT NULL,
+				syncronize_additional varchar(255) DEFAULT NULL,
+				enable_cart tinyint(1) DEFAULT NULL,
+				cart_autoresponder varchar(255) DEFAULT NULL,
+				cart_autoresponder_id int(10) DEFAULT NULL,
+				cart_delay int(10) DEFAULT NULL,
+				cart_cutoff int(10) DEFAULT NULL,
+				cart_options varchar(255) DEFAULT NULL,
+				PRIMARY KEY  (id)
+				) $charset_collate;";
 		dbDelta( $smaily );
 
+		// Create smaily_abandoned_cart table.
+		$abandoned_table_name = $wpdb->prefix . 'smaily_abandoned_carts';
+		$abandoned            = "CREATE TABLE $abandoned_table_name (
+				id int(11) NOT NULL AUTO_INCREMENT,
+				customer_id int(11) DEFAULT NULL,
```

Comment:
I think having both `id` and `customer_id` columns are not necessary, because customer can never have multiple abandoned carts. I would change the table structure to have only `customer_id` field that is PRIMARY (not auto increment). Does that make sense?